client/collector.go

```diff
@@ -0,0 +1,130 @@
+package client
+
+import (
+	"errors"
+
+	collectors "github.com/replicatedhq/replicated/gen/go/v1"
+	"github.com/replicatedhq/replicated/pkg/types"
+)
+
+func (c *Client) ListCollectors(appID string, appType string) ([]types.CollectorInfo, error) {
+	appType, err := c.GetAppType(appID)
+	if err != nil {
+		return nil, err
+	}
+
+	if appType == "platform" {
```

Comment:
both `ListCollectors` functions are identical. You don't need this split logic here.